rcl/test/rcl_add_custom_gtest.cmake

```diff
@@ -0,0 +1,81 @@
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if(rcl_add_custom_gtest_INCLUDED)
+  return()
+endif()
+set(rcl_add_custom_gtest_INCLUDED TRUE)
+
+# include CMake functions
+include(CMakeParseArguments)
+
+#
+# Custom macro for adding a gtest in rcl.
+#
+# It also takes all the arguments of ament_add_gtest as well as
+# INCLUDE_DIRS, LIBRARIES, and AMENT_DEPENDENCIES which are passed to
+# target_include_directories(), target_link_libraries(), and
+# ament_target_dependencies() respectively.
+#
+# :param target: the target name which will also be used as the test name
+# :type target: string
+#
+# @public
+#
+macro(rcl_add_custom_gtest target)
```

Comment:
Nothing in the body is doing anything which needs to be preserved in the caller context so this should become a function. And then you could ditch all leading underscore for readability.